recipes/repo.rb

```diff
@@ -265,6 +265,162 @@
   end
 end
 
+# Pulp setup
+pulp_data_directory = '/var/repos/.pulp'
+user 'pulp' do
+  manage_home true
+  comment 'Pulp content manager'
+  uid '1200'
+end
+group 'docker' do
+  append true
+  members ['pulp']
+  action [:manage]
+end
+directory pulp_data_directory do
+  owner 'pulp'
+  mode '0700'
+end
+directory "#{pulp_data_directory}/media" do
+  owner 'pulp'
+end
+
+cookbook_file "#{pulp_data_directory}/initialize.py" do
+  owner 'pulp'
+  source 'rpm_repo_init.py'
+  mode '0600'
+end
+
+package 'redis'
+service 'redis' do
+  action [:start, :enable]
+end
+cookbook_file '/etc/redis/redis.conf' do
+  source "redis.conf"
+  notifies :restart, 'service[redis]', :immediately
+end
+
+package 'postgresql'
+service 'postgresql' do
+  action [:start, :enable]
+end
+execute "Create pulp postgres user" do
+  command %(/usr/bin/psql -c "CREATE USER pulp WITH SUPERUSER LOGIN")
+  not_if %(/usr/bin/psql -tc "SELECT 1 from pg_user WHERE usename = 'pulp'" | grep -q 1)
+  user 'postgres'
+end
+execute "Create pulp postgres database" do
+  command %(/usr/bin/psql -c "CREATE DATABASE pulp OWNER pulp")
+  not_if %(/usr/bin/psql -tc "SELECT 1 FROM pg_database WHERE datname = 'pulp'" | grep -q 1)
+  user 'postgres'
+end
+
+
+directory "/usr/local/bin"
+remote_file "/usr/local/bin/systemd-docker" do
+  source 'https://github.com/subdavis/systemd-docker/releases/download/1.0.0/systemd-docker'
+  mode '0755'
+end
+cookbook_file "#{pulp_data_directory}/Dockerfile" do
+  source 'pulp/Dockerfile'
+end
+
+execute 'docker build -t pulp_image .' do
+  cwd pulp_data_directory
+end
+
+execute 'pulp_django_migration' do
+  command 'docker run --user 1200 --rm -v /var/run/postgresql:/var/run/postgresql pulp_image pulpcore-manager migrate --noinput'
+end
+
+execute 'pulp_collect_static' do
+  command "docker run --user 1200 --rm -v #{pulp_data_directory}:/var/repos/.pulp pulp_image pulpcore-manager collectstatic --noinput"
+end
+
+pulp_admin_password = data_bag_item('ros_buildfarm_pulp_admin_password', node.chef_environment)['password']
+execute 'set_pulp_admin_password' do
+  command "docker run --user 1200 --rm -v /var/run/postgresql:/var/run/postgresql pulp_image pulpcore-manager reset-admin-password -p '#{pulp_admin_password}'"
+end
+
+# TODO * Create gnupg directory for pulp
+
+execute 'systemctl daemon-reload' do
+  action :nothing
+end
+
+template "/etc/systemd/system/pulp-api-endpoint.service" do
+  source "pulp/pulp.service.erb"
+  variables Hash[
+    description: "Pulp API Endpoint",
+    after_units: %w(postgresql.service redis-server),
+    required_units: %w(postgresql.service redis-server),
+    docker_create_args: %(-u 1200 -v #{pulp_data_directory}:/var/repos/.pulp -v /var/run/postgresql:/var/run/postgresql -v /var/run/redis:/var/run/redis -p 24817:24817),
```

Comment:
```suggestion
    docker_create_args: %(-u 1200:1200 -v #{pulp_data_directory}:/var/repos/.pulp -v /var/run/postgresql:/var/run/postgresql -v /var/run/redis:/var/run/redis -p 24817:24817),
```

Reply:
Right. I had dropped the : separator because in the podman work `1200:1200` indicates run the process with uid 1200 inside the container mapped to uid 1200 on the host whereas in docker it indicates the usual uid:gid.